addons/hw_drivers/websocket_client.py

```diff
@@ -42,14 +42,14 @@ def on_message(ws, messages):
     """
     messages = json.loads(messages)
     for document in messages:
-        if (document['message']['type'] == 'print'):
+        if (document['message']['type'] == 'iot_action'):
             payload = document['message']['payload']
             if helpers.get_mac_address() in payload['iotDevice']['iotIdentifiers']:
-                #send box confirmation
                 for device in payload['iotDevice']['identifiers']:
                     if device['identifier'] in main.iot_devices:
-                        main.iot_devices[device["identifier"]]._action_default(payload)
-                        send_to_controller(payload['print_id'], device['identifier'])
+                        main.iot_devices[device["identifier"]].ws_action(payload)
```

Comment:
why to `ws_action()` and not directly to `action()`

Reply:
What do you mean ? Is it about the name 'ws_action" or are you asking why I don't call the _action_default() method ?

Reply:
you create a new function who just send to _action()
and the request that _action() receive comes from action() [1]

[1] https://github.com/odoo/odoo/blob/master/addons/hw_drivers/driver.py#L50